src/bot/checks.rs

```diff
@@ -0,0 +1,137 @@
+use poise::serenity_prelude::{ChannelId, Permissions, RoleId};
+use sqlx::{MySql, Pool};
+use std::env;
+use tracing::error;
+
+use crate::bot::Context;
+use crate::mysql_lib;
+
+/// Checks if the author has any of the specified roles
+///
+/// Will return false when:
+/// - the message was not send in a guild
+/// - the author doesn't have any of the roles
+pub async fn has_one_of_roles(ctx: Context<'_>, roles: Vec<RoleId>) -> bool {
+    if let Some(member) = ctx.author_member().await {
+        return member.roles.iter().any(|role| roles.contains(role));
+    }
+    false
+}
+
+/// Checks if the author has the specified role
+///
+/// Will return false when:
+/// - the message was not send in a guild
+/// - the author doesn't the role
+#[allow(dead_code)]
+pub async fn has_role(ctx: Context<'_>, role: RoleId) -> bool {
+    has_one_of_roles(ctx, vec![role]).await
+}
+
+/// Checks that the author doesn't have any the specified role
+///
+/// Will return true when:
+/// - the message was not send in a guild
+/// - the author doesn't have any of the role
+#[allow(dead_code)]
+pub async fn has_not_roles(ctx: Context<'_>, roles: Vec<RoleId>) -> bool {
+    !has_one_of_roles(ctx, roles).await
+}
```

Comment:
grammar in the doc comment, plus the name of the function isn't really descriptive of what it does (`has_none_of_roles` maybe?)